### PR TITLE
added AdamW and SGDW and corresponding schedular

### PR DESCRIPTION
**Pull request by kashif**
for issue #220 and #228

### CODE REVIEW

**Comment by alanakbik** [comment]
Awesome!

File: flair/trainers/language_model_trainer.py [hunk 1]
```diff
@@ -186,9 +191,15 @@ def train(self,
 
             epoch = 0
             best_val_loss = self.model.best_score if self.model.best_score is not None else 100000000
-            optimizer = torch.optim.SGD(self.model.parameters(), lr=learning_rate)
-            scheduler: ReduceLROnPlateau = ReduceLROnPlateau(optimizer, verbose=True, factor=anneal_factor,
-                                                             patience=patience)
+            optimizer = self.optimzer(self.model.parameters(), lr=learning_rate, **kwargs)
+            if optimizer.__class__.__name__ in ['AdamW', 'SGDW']:
```
Comment (tabergma) [inline flair/trainers/language_model_trainer.py]: Due to consistency I would prefer `type(optimizer) is AdamW or type(optimizer) is SGDW`. Same for the other occurrences.

**Comment by kashif** [comment]
🙇

File: flair/trainers/language_model_trainer.py (hunk 1 above)
Comment (kashif) [reply flair/trainers/language_model_trainer.py]: Can I do:

```python
if isinstance(optimizer, (AdamW, SGDW)):
    ...
```

**Pull request merged**